BgsService/BgsFactionData.cs

```diff
@@ -83,10 +83,15 @@ private static Faction ParseFaction(object response)
             {
                 IDictionary<string, object> presenceJson = (IDictionary<string, object>)presence;
                 FactionPresence factionPresence = new FactionPresence()
+                /*
+                factionPresence.systemName = (string)presenceJson["system_name"];
+                factionPresence.influence = (decimal?)(double?)presenceJson["influence"] * 100;
+                factionPresence.FactionState = FactionState.FromEDName((string)presenceJson["state"]) ?? FactionState.None;
+                */
                 {
-                    systemName = (string)presenceJson["system_name"],
-                    influence = (decimal?)(double?)presenceJson["influence"] * 100, // Convert from a 0-1 range to a percentage
-                    FactionState = FactionState.FromEDName((string)presenceJson["state"]) ?? FactionState.None,
+                    systemName = JsonParsing.getString(presenceJson, "system_name"),
+                    influence = (JsonParsing.getOptionalDecimal(presenceJson, "influence") ?? 0) * 100, // Convert from a 0-1 range to a percentage
```

Comment:
I prefer writing `100M` to make it a literal decimal but it's just a style point.